### PR TITLE
linter check full command group names

### PR DESCRIPTION
**Pull request by kairu-ms**
Currently azure cli cannot detect the miss of some command group help, because they are not as keys of command_group_table. This PR fix the issue.

### CODE REVIEW

File: azdev/operations/linter/linter.py [hunk 1]
```diff
@@ -58,7 +58,17 @@ def commands(self):
...
-        return self._command_loader.command_group_table.keys()
+        if not self._command_groups:
+            added_command_groups = set()
+            for command_group in self._command_loader.command_group_table.keys():
+                prefix_name = ""
+                for word in command_group.split():
+                    prefix_name = "{} {}".format(prefix_name, word).strip()
+                    if prefix_name in added_command_groups:
+                        continue
+                    added_command_groups.add(prefix_name)
```
Comment (Juliehzl) [inline azdev/operations/linter/linter.py]: Is it possible to add some comment here for easy understanding?

File: azdev/__init__.py [hunk 2]
```diff
@@ -4,4 +4,4 @@
 # license information.
 # -----------------------------------------------------------------------------
 
-__VERSION__ = '0.1.33'
+__VERSION__ = '0.1.34'
```
Comment (Juliehzl) [inline azdev/__init__.py:7]: also need to add release note?

File: azdev/operations/linter/linter.py [hunk 3]
```diff
@@ -58,7 +58,17 @@ def commands(self):
...
     def command_groups(self):
-        return self._command_loader.command_group_table.keys()
+        if not self._command_groups:
+            added_command_groups = set()
+            for command_group in self._command_loader.command_group_table.keys():
+                prefix_name = ""
+                for word in command_group.split():
+                    prefix_name = "{} {}".format(prefix_name, word).strip()
+                    if prefix_name in added_command_groups:
+                        continue
```
Comment (kairu-ms) [inline azdev/operations/linter/linter.py]:
```suggestion
                    if prefix_name in added_command_groups:
                        # if the parent command group is added continue
                        continue
```

File: azdev/operations/linter/linter.py [hunk 4]
```diff
@@ -139,6 +150,8 @@ def command_group_expired(self, command_group_name):
             deprecate_info = group_kwargs.get('deprecate_info', None)
             if deprecate_info:
                 return deprecate_info.expired()
+        except KeyError:
```
Comment (Juliehzl) [inline azdev/operations/linter/linter.py:153]: could you add comment for the scenario we get `KeyError`?

Comment (kairu-ms) [reply azdev/operations/linter/linter.py:153]:
```suggestion
        # ignore command_group_name which is not in command_group_table.
        except KeyError:
```

File: azdev/operations/linter/linter.py [hunk 5]
```diff
@@ -139,6 +150,9 @@ def command_group_expired(self, command_group_name):
             deprecate_info = group_kwargs.get('deprecate_info', None)
             if deprecate_info:
                 return deprecate_info.expired()
+        # ignore command_group_name which is not in command_group_table.
+        except KeyError:
+            pass
```
Comment (kairu-ms) [inline azdev/operations/linter/linter.py]:
```suggestion
        except KeyError:
            # ignore command_group_name which is not in command_group_table.
            pass
```

**Review by Juliehzl — APPROVED** [review]

File: HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.1.34
+++++++
+* `azdev linter`: support to detect commmand groups which are missing in command_group_table.
```
Comment (jiasli) [inline HISTORY.rst:8]: PR number should be added to release note.

**Comment by jiasli** [comment]
To add more context to the PR:

Per my understanding, if only `ad app credential` exists in `command_group_table`, `azdev linter` only checks

- ✔ `ad app credential`

but not

- ❌ `ad`
- ❌ `ad app`

This PR makes `azdev` check `ad` and `ad app` as well - it checks all

- ✔ `ad`
- ✔ `ad app`
- ✔ `ad app credential`

**Pull request merged**